### DISCUSSION TITLE
Compiler Version Error

### DISCUSSION THREAD

**Discussion by ChuckMurr** [Cyfrin/foundry-full-course-cu#35] [Q&A] [answered]
I am using the Solidity extension by Juan Blanco. I am attempting to find a way to automatically switch compiler versions between each file and project. I have found solutions where I have to hardcode the compiler versions in or where I have to manually switch compiler versions every single time I want to compile a different file. Can anyone point me in the right direction? I've been searching for a while at this point. Thanks!

**Comment by cromewar** [comment, marked as answer]
You can change the extension to the one provided by hardhat, or use similar compiler versions on your contracts. For example:
If you setup the compiler version as 0.8.9 it will use the exact version.
If you setup the compiler version as `^0.8.9` this will accept any version from this one to the next newer one, lets suppose a hypothetical version 0.9.

So using `^` is a good solution.

**Comment by ChuckMurr** [comment]
Thanks for the help!
